src/Text/XML/DSig.hs

```diff
@@ -183,9 +194,10 @@ mkSignCredsWithCert mValidSince size = do
 -- list of 'SignCred's.
 {-# NOINLINE verify #-}
 verify :: forall m. (MonadError String m) => NonEmpty SignCreds -> LBS -> String -> m ()
-verify creds el signedID = case unsafePerformIO (verifyIO creds el signedID) of
-  []   -> pure ()
-  errs -> throwError $ show (snd <$> errs)
+verify creds el signedID = case unsafePerformIO (try $ verifyIO creds el signedID) of
```

Comment:
By the way, why is it safe to use `unsafePerformIO` here? (Not familiar with what `verifyIO` does)

Reply:
4c6e74fdef63db24e66fbe88656afe8833c72ab8